src/components/ui/ShareMenu.jsx

```diff
@@ -0,0 +1,115 @@
+/* global _ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Telemetry from '../../libs/telemetry';
+
+const facebookShareUrl = location => {
+  return `https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(location)}`;
+};
+
+const twitterShareUrl = location => {
+  return `https://twitter.com/intent/tweet?url=${ encodeURIComponent(location) }`;
+};
+
+const menu_height = 3 * 32;
+
+export default class ShareMenu extends React.Component {
+  _isMounted = false;
+
+  static propTypes = {
+    url: PropTypes.string.isRequired,
+    scrollableParent: PropTypes.string.isRequired,
+  }
+
+  state = {
+    open: false,
+    top: 0,
+    left: 0,
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
```

Comment:
We need to call `this.close()` here to prevent the following error when navigating in the panel while a menu is still opened: `react_devtools_backend.js:6 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.`